applications/nrf5340_audio/src/bluetooth/bt_content_control/content/bt_media_control.c

```diff
@@ -17,21 +17,68 @@
 #include "hw_codec.h"
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_REGISTER(ble_audio_services, CONFIG_AUDIO_SERVICES_LOG_LEVEL);
+LOG_MODULE_REGISTER(bt_media_control, CONFIG_BT_MEDIA_CONTROL_LOG_LEVEL);
 
 static uint8_t media_player_state = BT_MCS_MEDIA_STATE_PLAYING;
 
 static struct media_player *local_player;
-static ble_mcs_play_pause_cb play_pause_cb;
+static bt_media_control_play_pause_cb play_pause_cb;
 
-static uint8_t mcp_mcs_disc_status[CONFIG_BT_MAX_CONN];
+struct media_ctlr {
+	uint8_t mcp_mcs_disc_status;
+	struct bt_conn *conn;
+};
+
+static struct media_ctlr mcc_peer[CONFIG_BT_MAX_CONN];
 
 enum mcs_disc_status {
 	IDLE,
 	IN_PROGRESS,
 	FINISHED,
 };
 
+/**
+ * @brief	Get the index of the first available mcc_peer
+ *
+ * @return	Index if success, -ENOMEM if no available indexes
+ */
+static int mcc_peer_index_free_get(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(mcc_peer); i++) {
+		if (mcc_peer[i].conn == NULL) {
+			return i;
+		}
+	}
+
+	LOG_WRN("No more indexes for MCC peer");
+
+	return -ENOMEM;
+}
+
+/**
+ * @brief	Get index of a given conn pointer
+ *
+ * @param	conn	Pointer to check against
+ *
+ * @return	index if found, -ESRCH if not found, -EINVAL if invalid conn pointer
+ */
+static int mcc_peer_index_get(struct bt_conn *conn)
+{
+	if (conn == NULL) {
+		LOG_WRN("Invalid conn pointer");
+		return -EINVAL;
+	}
+
+	for (uint8_t i = 0; i < ARRAY_SIZE(mcc_peer); i++) {
+		if (mcc_peer[i].conn == conn) {
+			return i;
+		}
+	}
+
+	LOG_DBG("No matching conn pointer for this mcc_peer");
+	return -ESRCH;
+}
```

Comment:
Have you considered using the same index as `conn` has? i.e. use `bt_conn_index` to store and retrieve from `mcc_peer`, instead of doing loops?

Reply:
We used to use bt_conn_index, but when we have a disconnect there is no longer a valid conn-pointer, causing an assert

Reply:
> but when we have a disconnect there is no longer a valid conn-pointer, causing an assert

On disconnect the pointer does not become invalid. You can call `bt_conn_index` on a disconnected `conn` object just fine, so it would only be if you change the pointer to `NULL` that would be an issue.

Reply:
I did test this, and somewhere in host it was set to NULL, because the conn pointer given to disconnected_cb was not valid once I got to bt_media_control_conn_disconnected

Reply:
Right, there may be functions (or rather callbacks) where the `conn` pointer supplied by the host will be NULL, in which cases you'd need some other way of looking up things. That makes sense :) Of course that does not really fix this one, as you won't do a lookup anyways if `conn == NULL`, so you could still depend on `bt_conn_index` for all lookups, since you won't be able to lookup with `conn == NULL` regardless

Reply:
> The conn pointer is not NULL, but the bt_conn_index asserts because the index is probaly negative:
> 
> ```c
> index = conn - acl_conns;
> __ASSERT(index >= 0 && index < ARRAY_SIZE(acl_conns), "Invalid bt_conn pointer");
> ```

That can only happen if the `conn` pointer privided isn't a valid pointer. 

The connection objects are statically allocated. Once you get a pointer, it will be valid for the lifetime of the OS; it will never become invalid. This can only assert if you provide `bt_conn_index` with an invalid pointer (i.e. not a pointer provided by the host). 

In the above, if `index` becomes negative, it means that the `conn` pointer provided by the application, does not lie within `acl_conns`.